lib/hyperion/result_handling/result_maker.rb

```diff
@@ -35,6 +36,8 @@ def make_from_typho(typho_result, result_class)
       read_body = proc { read(typho_result.body, route.response_descriptor) }
       code = typho_result.code
 
+      log_error_response(read(typho_result.body, :json)) unless typho_result.success?
```

Comment:
Need tests for this. What happens if the body is not JSON? Are you sure this works properly for all the non-success cases? timed out? no response? bad route? client error? server error? check code?